src/helpers/courses.js

```diff
@@ -195,73 +195,113 @@ const listForOperations = async (query, origin, credentials) => {
 };
 
 const listForPedagogy = async (query, credentials) => {
-  const traineeId = query.trainee || get(credentials, '_id');
-  const trainee = await User.findOne({ _id: traineeId }).lean();
+  const traineeOrTutorId = query.trainee || get(credentials, '_id');
   const shouldQueryCompanies = !!query.holding || !!query.company;
   const companies = [];
   if (query.holding) companies.push(...credentials.holding.companies);
   if (query.company) companies.push(query.company);
 
   const courses = await Course.find(
     {
-      trainees: trainee._id,
-      $or: [
+      $and: [
```

Comment:
Ce and est necessaire ?

Reply:
comment tu l'écrirais sans $and ? tu vois une façon de faire sans avoir 2 champs $or ?

Reply:
Tu as raison, on veut que nos conditions soient toutes verifiees, il faut donc garder le $and